bool top
bool bot
float32 top_x
float32 top_y
float32 bot_x
float32 bot_y